### PR TITLE
fix #4615: chown isnt showing a message

### LINKED ISSUE #4615: `chown -v --from=42 43` isn't showing a message
With GNU:
```
$ touch f
$ chown -v --from=42 43 f
ownership of 'f' retained as sylvestre
```

Nothing is showed with our implementation

To run it:
` fakeroot bash util/run-gnu-test.sh   tests/chown/basic.sh `

### PR DESCRIPTION
**Pull request by djedi23**
This PR add a message when the current user is not matched by the `from` argument and the verbosity is increased.
Two different messages are displayed depending if a group is set or not to match the observed behavior on GNU chown.

This PR fixes  #4615.

### CODE REVIEW

**Comment by sylvestre** [comment]
Seems that the GNU test isn't fixed:
```

FAIL: tests/chown/basic
=======================

chown: ownership of 'f' retained as root
--- exp	2023-04-21 06:05:45.448369817 +0000
+++ out	2023-04-21 06:05:45.444369729 +0000
@@ -1 +0,0 @@
-ownership of 'f' retained as root
make[5]: *** [Makefile:21405: tests/test-suite-root.log] Error 1
chown: cannot dereference 'nf': No such file or directory
--- exp	2023-04-21 06:05:45.452369904 +0000
+++ out	2023-04-21 06:05:45.448369817 +0000
@@ -1 +0,0 @@
-failed to change ownership of 'nf' to 0
FAIL tests/chown/basic.sh (exit status: 1)

```

**Comment by djedi23** [comment]
I'm not reproducing the test failure on my workstation. Maybe I missed an option or something in a configuration.
```sh
$ fakeroot bash util/run-gnu-test.sh   tests/chown/basic.sh
ME_dir='/home/djedi/prj/coreutils/util'
REPO_main_dir='/home/djedi/prj/coreutils'
path_UUTILS='/home/djedi/prj/coreutils'
path_GNU='/home/djedi/prj/gnu'
[ pwd:'/home/djedi/prj/gnu' ]
Running specific tests: tests/chown/basic.sh
make  check-recursive
make[1]: Entering directory '/home/djedi/prj/gnu'
Making check in .
make[2]: Entering directory '/home/djedi/prj/gnu'
make  check-TESTS check-local
make[3]: Entering directory '/home/djedi/prj/gnu'
  GEN      check-README
  GEN      check-duplicate-no-install
  GEN      sc-avoid-builtin
  GEN      sc-avoid-io
  GEN      sc-avoid-non-zero
  GEN      sc-avoid-path
  GEN      sc-avoid-timezone
  GEN      sc-avoid-zeroes
  GEN      sc-exponent-grouping
  GEN      sc-lower-case-var
  GEN      check-texinfo
make[4]: Entering directory '/home/djedi/prj/gnu'
PASS: tests/chown/basic.sh
============================================================================
Testsuite summary for GNU coreutils 9.3
============================================================================
# TOTAL: 1
# PASS:  1
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[4]: Leaving directory '/home/djedi/prj/gnu'
make[3]: Leaving directory '/home/djedi/prj/gnu'
make[2]: Leaving directory '/home/djedi/prj/gnu'
make[1]: Leaving directory '/home/djedi/prj/gnu'
```

If I try to reproduce the whole test suite, everything seems passing:
```sh
$ CI=1 bash util/run-gnu-test.sh run-root
ME_dir='/home/djedi/prj/coreutils/util'
REPO_main_dir='/home/djedi/prj/coreutils'
path_UUTILS='/home/djedi/prj/coreutils'
path_GNU='/home/djedi/prj/gnu'
[ pwd:'/home/djedi/prj/gnu' ]
Running specific tests: run-root
Running check-root to run only root tests
make check TESTS='tests/chown/basic.sh tests/cp/cp-a-selinux.sh tests/cp/preserve-gid.sh tests/cp/special-bits.sh tests/cp/cp-mv-enotsup-xattr.sh tests/cp/capability.sh tests/cp/sparse-extents-2.sh tests/cp/cross-dev-symlink.sh tests/dd/skip-seek-past-dev.sh tests/df/problematic-chars.sh tests/df/over-mount-device.sh tests/du/bind-mount-dir-cycle.sh tests/du/bind-mount-dir-cycle-v2.sh tests/id/setgid.sh tests/install/install-C-root.sh tests/ls/capability.sh tests/ls/nameless-uid.sh tests/misc/chcon.sh tests/misc/chroot-credentials.sh tests/misc/selinux.sh tests/misc/truncate-owned-by-other.sh tests/mkdir/writable-under-readonly.sh tests/mkdir/smack-root.sh tests/mv/hardlink-case.sh tests/mv/sticky-to-xpart.sh tests/rm/fail-2eperm.sh tests/rm/no-give-up.sh tests/rm/one-file-system.sh tests/rm/read-only.sh tests/rm/empty-immutable-skip.sh tests/tail-2/append-only.sh tests/tail-2/end-of-device.sh tests/touch/now-owned-by-other.sh' SUBDIRS=.
make[1]: Entering directory '/home/djedi/prj/gnu'
make  check-recursive
make[2]: Entering directory '/home/djedi/prj/gnu'
Making check in .
make[3]: Entering directory '/home/djedi/prj/gnu'
make  check-TESTS check-local
make[4]: Entering directory '/home/djedi/prj/gnu'
  GEN      check-README
  GEN      check-duplicate-no-install
  GEN      sc-avoid-builtin
  GEN      sc-avoid-io
  GEN      sc-avoid-non-zero
  GEN      sc-avoid-path
  GEN      sc-avoid-timezone
  GEN      sc-avoid-zeroes
  GEN      sc-exponent-grouping
  GEN      sc-lower-case-var
  GEN      check-texinfo
make[5]: Entering directory '/home/djedi/prj/gnu'
cp-a-selinux.sh: skipped test: this system lacks SELinux support
SKIP: tests/cp/cp-a-selinux.sh
PASS: tests/chown/basic.sh
PASS: tests/cp/special-bits.sh
capability.sh: skipped test: configured without libcap support
SKIP: tests/cp/capability.sh
PASS: tests/cp/preserve-gid.sh
PASS: tests/dd/skip-seek-past-dev.sh
cp-mv-enotsup-xattr.sh: skipped test: failed to get xattr of file
SKIP: tests/cp/cp-mv-enotsup-xattr.sh
PASS: tests/cp/cross-dev-symlink.sh
PASS: tests/du/bind-mount-dir-cycle.sh
PASS: tests/df/problematic-chars.sh
PASS: tests/du/bind-mount-dir-cycle-v2.sh
PASS: tests/id/setgid.sh
capability.sh: skipped test: configured without libcap support
SKIP: tests/ls/capability.sh
PASS: tests/df/over-mount-device.sh
PASS: tests/ls/nameless-uid.sh
PASS: tests/install/install-C-root.sh
chcon.sh: skipped test: this system lacks SELinux support
SKIP: tests/misc/chcon.sh
selinux.sh: skipped test: this system lacks SELinux support
SKIP: tests/misc/selinux.sh
writable-under-readonly.sh: skipped test: temporarily disabled
PASS: tests/misc/truncate-owned-by-other.sh
SKIP: tests/mkdir/writable-under-readonly.sh
PASS: tests/misc/chroot-credentials.sh
smack-root.sh: skipped test: this system lacks SMACK support
SKIP: tests/mkdir/smack-root.sh
hardlink-case.sh: skipped test: failed to create hfs file system
SKIP: tests/mv/hardlink-case.sh
PASS: tests/mv/sticky-to-xpart.sh
PASS: tests/rm/fail-2eperm.sh
PASS: tests/rm/no-give-up.sh
PASS: tests/rm/empty-immutable-skip.sh
PASS: tests/rm/one-file-system.sh
PASS: tests/rm/read-only.sh
PASS: tests/tail-2/end-of-device.sh
PASS: tests/touch/now-owned-by-other.sh
PASS: tests/cp/sparse-extents-2.sh
PASS: tests/tail-2/append-only.sh
============================================================================
Testsuite summary for GNU coreutils 9.3
============================================================================
# TOTAL: 33
# PASS:  24
# SKIP:  9
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[5]: Leaving directory '/home/djedi/prj/gnu'
make[4]: Leaving directory '/home/djedi/prj/gnu'
make[3]: Leaving directory '/home/djedi/prj/gnu'
make[2]: Leaving directory '/home/djedi/prj/gnu'
make[1]: Leaving directory '/home/djedi/prj/gnu'
```

File: src/uucore/src/lib/features/perms.rs [hunk 1]
```diff
@@ -260,6 +260,27 @@ impl ChownExecutor {
                 }
             }
         } else {
+            if self.verbosity.level == VerbosityLevel::Verbose {
```
Comment (sylvestre) [inline src/uucore/src/lib/features/perms.rs:263]: Maybe move this into a function?

```
fn print_verbose(&self, path: &Path, uid: u32, gid: Option<u32>) {
    if self.verbosity.level == VerbosityLevel::Verbose {
        match gid {
            Some(gid) => {
                println!(
                    "ownership of {} retained as {}:{}",
                    path.quote(),
                    entries::uid2usr(uid).unwrap_or_else(|_| uid.to_string()),
                    entries::gid2grp(gid).unwrap_or_else(|_| gid.to_string()),
                );
            }
            None => {
                println!(
                    "ownership of {} retained as {}",
                    path.quote(),
                    entries::uid2usr(uid).unwrap_or_else(|_| uid.to_string()),
                );
            }
        }
    }
}

```

and call:
`print_verbose(path, uid, self.dest_gid.map(|_| meta.gid()));`

**Comment by cakebaker** [comment]
With GNU chown 9.3, the following commands:
```
$ chown -v --from=42 :43 f
$ chown -v --from=:42 :43 f
$ chown -v --from=42:42 :43 f
```
all output:
```
group of 'f' retained as dho
```

uutils chown, on the other hand, returns in all three cases:
```
ownership of 'f' retained as dho:dho
```

**Comment by djedi23** [comment]
Corrected.

**Comment by cakebaker** [comment]
Thanks!

**Pull request merged**